sale_coupon_product_management/models/sale_coupon_program_option.py

```diff
@@ -114,6 +121,7 @@ def get_program_values(self):
         for option in self:
             opt_cfg = cfg[option.option_type]
             opt_vals = opt_cfg["values"]
+            option._update_values_from_program(opt_vals, program=program)
```

Comment:
then
```suggestion
            opt_vals.update(option._update_values_from_program(program))
```

Reply:
Well in this case it would complicate things, because this is not single level dictionary. It can be multi level dictionary, which then would need to be built inside `_update_values_from_program` (using template), then updated and when return, merged with existing opt_vals.

So this would complicate things. If I just update single value, I dont need to do any of that and main method (`get_program_values`) does not need to be even ware how it is done.

Reply:
You just have to return `{DISCOUNT_PRODUCT_FNAME: {"list_price": program.discount_fixed_amount}}`.
Or maybe there's something I don't see?

Reply:
You can't, because if you would use it on `update`, it would overwrite existing dict on `DISCOUNT_PRODUCT_FNAME` key.  That dict might be already prefilled with other values (thus why there is multi-level dict).

As mentioned by TODO: its possible to use `mergedeep` module that can help, but still it complicates this, cause we would then merge dictionaries twice. In this case and after.

Reply:
In other words you want to either merge dictionaries (not update) or update one specific value -> which in my opinion looks simpler (if we would need to update many key/values, then merging might be more beneficial).

Reply:
then you could maybe do it like this : `opt_vals[DISCOUNT_PRODUCT_FNAME].update(…)`

Reply:
But then main method starts doing more then it should. its responsibility for `_update_values_from_program` and it should be able to update not just whats on `DISCOUNT_PRODUCT_FNAME` (it so happens that in this case we now update only one value there, but it could be that we would need to update more. And then it would mean multiple `update` calls)